### ISSUE TITLE
JFilechooser look and feel not updated when change themes in runtime.

### ISSUE THREAD

**Issue by MXandresMX** [JFormDesigner/FlatLaf#619]
If we create a variable of JFilechooser like this

`private final JFileChooser chooserFileBackup = new JFileChooser();`

and first setup theme like dark and later setup the theme light  when we show the dialog like this

`if (chooserFileBackup.showOpenDialog(this) == JFileChooser.APPROVE_OPTION) {}`

we will see the theme light   was not applied in all the dialog frame check this picture .

[https://ibb.co/YXhzZTc](url)

That not happen if we create at runtime each time 
JFileChooser chooserFileBackup = new JFileChooser();

But that is not the case.

For fix it we need to call in this way 
`
 chooserFileBackup.updateUI();
if (chooserFileBackup.showOpenDialog(this) == JFileChooser.APPROVE_OPTION) {}
`
 Just in that case the Dialog is updated correctly.

**Comment by DevCharly** [comment]
This is normal behavior.

`FlatLaf.updateUI()` does not know components that you remember somewhere. It updates only components that are contained in windows.

To fix/avoid this, either invoke `SwingUtilities.updateComponentTreeUI( component )` yourself (if necessary) or do not store components anywhere. Instead create them when needed.

**Comment by MXandresMX** [comment]
I understand thanks.

**Comment by MXandresMX** [comment]
Talking about the same for example if we have a frame with JXHeader when we change FONT in runtime the JXHeader don't change the font. 

Other components without problems.

**Comment by DevCharly** [comment]
For this case, you have to override the `updateUI()` method and update the font there.

E.g. if you use something like this:
~~~java
class MyPanel extends JPanel {
  private JXHeader header;

  MyPanel() {
    header = new JXHeader();
    header.setTitleFont( myFont );
    add( header );
  }
}
~~~

Then you have to add a `updateUI()` method, which is invoked when theme changes:
~~~java
class MyPanel extends JPanel {
  private JXHeader header;

  MyPanel() {
    header = new JXHeader();
    header.setTitleFont( myFont );
    add( header );
  }

  @Override
  public void updateUI() {
    super.updateUI();

    if( header == null )
      return; // header not yet created when invoked from JPanel constructor

    header.setTitleFont( myFont );
  }
}
~~~

**Comment by MXandresMX** [comment]
Since i use a simple JPanel and not custom i can't use updateUI();
So, i fixed the problem using example **FlatDarkLaf.properties**

```
#Default + 2
JXHeader.titleFont = $large.font
#Default FAIL
JXHeader.descriptionFont = $defaultFont
#Default WORK
JXHeader.descriptionFont = 0
```

I want to use default font from the java app. But i have a dude, i searched in Typography [Typography](https://www.formdev.com/flatlaf/typography/) **UI Key** i can't use **$defaultFont** I get this error, but other UI Key are working fine.

`Exception in thread "AWT-EventQueue-0" java.lang.IllegalArgumentException: variable or property 'defaultFont' not found`

**Comment by DevCharly** [comment]
Oh, sorry. I misunderstood you.

I'll fix this in FlatLaf SwingX addon.
There are more SwingX components that have the same problem...

Regarding using `$defaultFont` in properties files: this is usually not possible because it is not defined in properties files in core themes.
So using `JXHeader.descriptionFont = 0` or `JXHeader.descriptionFont = +0` is best.

**Comment by DevCharly** [comment]
SwingX font issue is fixed in commit 9fef2f9d0514a8391e0a091b1370a20a1a69e309
